cloudinit/config/cc_set_hostname.py

```diff
@@ -21,9 +21,14 @@
 the ``fqdn`` config key. If both ``fqdn`` and ``hostname`` are set, ``fqdn``
 will be used.
 
+In order to correctly set the hostname, cloud-init will always run this module
+if metadata from a local source is already received specifying a hostname
```

Comment:
I think we can simplify this sentence a little, but perhaps also add an example or two to help people understand when this might happen:

```suggestion
if metadata from a local source (e.g. ???) specifies a hostname
```

Do we have documentation of "local" vs not-"local" sources somewhere? A link to that would seem appropriate here too.